### PR TITLE
[13.0][IMP] stock_picking_invoice_link: Better link invoices to stock_moves process

### PR DESCRIPTION
**Pull request by carlosdauden**
Steps to reproduce the issue before this PR:
 - Invoice sale order
 - Create Credit note whith third option
 - Remove new draft invoice
 - Invoice sale order
 - Print invoice

@Tecnativa TT27156

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please check Travis

**Comment by carlosdauden** [comment]
Travis green

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
These complex cases need a unit test exercising the problem that is solving. On contrary, regressions are very likely to happen on migrations or refactorings.

I understand the the aim of this is to allow to link stock move lines in refunds, isn't it? Then put that in a complete test.

**Comment by carlosdauden** [comment]
I agree.
We improve the tests before merging

**Comment by pedrobaeza** [comment]
Any news on this?

**Comment by pedrobaeza** [comment]
@carlosdauden @sergio-teruel any news on this?

**Comment by pedrobaeza** [comment]
@carlosdauden @sergio-teruel any news on this?

File: stock_picking_invoice_link/models/sale_order.py [hunk 1]
```diff
@@ -38,8 +48,6 @@ def _prepare_invoice_line(self):
             )
             < 0
         ):
-            stock_moves = stock_moves.filtered(
-                lambda m: m.to_refund and not m.invoice_line_ids
-            )
+            stock_moves = stock_moves.filtered(lambda m: m.to_refund)
```
Comment (pedrobaeza) [inline stock_picking_invoice_link/models/sale_order.py]:
```suggestion
            stock_moves = stock_moves.filtered("to_refund")
```

**Comment by CarlosRoca13** [comment]
Added test for new use case and change done @pedrobaeza

**Review by CarlosRoca13 — APPROVED** [review]

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
/ocabot merge patch

Please forward port it to v14

**Pull request merged**